internal/pkg/csumreader/csumreader.go

```diff
@@ -0,0 +1,91 @@
+package csumreader
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+const (
+	ChecksumLen = 8
+)
+
+var (
+	ErrChecksumMismatch = errors.New("checksum mismatch")
+)
+
+type ChecksumReader struct {
+	dataFile              io.Reader
+	checksumFile          io.Reader
+	buf                   []byte
+	offset                int
+	chunkSize             int
+	eof                   bool
+	disableChecksumVerify bool
```

Comment:
How about renaming this argument to enableChecksumVerify? Then we can remove the double-negative `if` statment in L57.